confmap/confmap.go

```diff
@@ -24,6 +24,12 @@ const (
 	KeyDelimiter = "::"
 )
 
+const (
+	// MapstructureTag is the struct field tag used to record marshaling/unmarshaling settings.
+	// See https://pkg.go.dev/github.com/go-viper/mapstructure/v2 for supported values.
+	MapstructureTag = "mapstructure"
```

Comment:
We could just hardcode the tag name in `xconfmap` for now, but I figure it's a good time to make this an exported constant.

Reply:
We could also move this to an `confmap/internal` package I guess, if we want to avoid exposing this